python/context_selector/context_widget.py

```diff
@@ -0,0 +1,956 @@
+        :param str link_display_override: Override text to be displayed for the link.
+        """
+        logger.debug("Setting context to: %s" % (context,))
+
+        # clear any related tasks from the previous context
+        self._menu_actions["Related"] = []
+
+        self._context = context
+        self._show_context(
+            context,
+            task_display_override=task_display_override,
+            link_display_override=link_display_override
+        )
+
+        # ensure the new context is added to the list of recents.
+        # TODO: we should only update recents after publishing. but until we
+        # have the ability to easily assign a context to multiple publish items
+        # simultaneously, this provides a nice mechanism for quick assignment
+        # in the UI.
+        if context:
+            self._add_to_recents(context)
+
+    def set_up(self, task_manager):
+        """
+        Handles initial set up of the widget. Includes setting up menu, running
+        any background set up tasks, etc.
+
+        :param task_manager: Background task manager to use
+        :type task_manager: :class:`~tk-framework-shotgunutils:task_manager.BackgroundTaskManager`
+        """
+        logger.debug("Setting up the UI...")
+
+        self._task_manager = task_manager
+
+        # attach the context menu
+        self.ui.task_menu_btn.setMenu(self._task_menu)
+        self._task_menu.aboutToShow.connect(
+            self._on_about_to_show_contexts_menu)
+
+        # setup the search toggle
+        self.ui.task_search_btn.toggled.connect(self._on_task_search_toggled)
+        self.ui.task_search.hide()
+
+        # setup the search toggle
+        self.ui.link_search_btn.toggled.connect(self._on_link_search_toggled)
+        self.ui.link_search.hide()
+
+        # set up the task manager to the task search widget
+        self.ui.task_search.set_placeholder_text("Search for Tasks...")
+        self.ui.task_search.set_bg_task_manager(task_manager)
+        self.ui.task_search.entity_activated.connect(
+            self._on_entity_activated)
+
+        # save as above but for the link search widget
+        self.ui.link_search.set_placeholder_text("Search for entity link...")
+        self.ui.link_search.set_bg_task_manager(task_manager)
+        self.ui.link_search.entity_activated.connect(
+            self._on_entity_activated
+        )
+
+        # set up event filters for the task/link display labels so that when
+        # clicked, they go directly to an edit state
+        self.ui.task_display.installEventFilter(self)
+        self.ui.link_display.installEventFilter(self)
+
+        # setup event filters for the task/link search inputs so that when
+        # certain keys are pressed, the widget can react to it properly
+        self.ui.task_search.installEventFilter(self)
+        self.ui.link_search.installEventFilter(self)
+
+        # we need to limit the search completer to entity types that are valid
```

Comment:
Should we reconsider this logic since this is no longer limited to the publisher?

Reply:
new methods added!